auth.go

```diff
@@ -184,6 +184,12 @@ func (a Authenticator) Exchange(code string) (*oauth2.Token, error) {
 	return a.config.Exchange(a.context, code)
 }
 
+// ExchangeWithOpts performs the same function as the Exchange function
+// but takes in optional URL Auth params
+func (a Authenticator) ExchangeWithOpts(code string, opts ...oauth2.AuthCodeOption) (*oauth2.Token, error) {
```

Comment:
Could we just add this variadic parameter to the main Exchange() method? It'll be non-breaking for existing users (since no options is valid) and keep the overall interface simpler.

Reply:
I'm ok with it.
I was just trying to mimic the behavior with Token and TokenWithOpts funcs.

Let me change it.